tests/src/test/java/ca/bc/gov/open/jag/efiling/page/PackageReviewPage.java

```diff
@@ -95,11 +95,11 @@
         List<String> packageDetails = new ArrayList<>();
         packageValueDetails.forEach(webElement -> {
             if (!webElement.isDisplayed()) {
-                logger.info("WebElement is still not visible. Waiting for all rows to be present: {}", webElement.isDisplayed());
+                logger.info("WebElement is still not visible. Waiting for all rows to be present: {}", Boolean.valueOf(webElement.isDisplayed()));
                 wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath("//div[@data-test-id='detailsTable'] //*[contains(@class,'bcgov-table-value')]/b")));
             }
-            logger.info("Is it visible: {}", webElement.isDisplayed());

```

Comment:
I'm not sure this is necessary in parameterized logging.

Reply:
You may be right, however, it's still autoboxing.  logger.info() is accepting a String and an Object (a java.lang.Boolean in this case) but webElement.isDisplayed() is returning a little b boolean, primitive, not an Object.  Java is autoboxing, auto-correcting the code, and converts the boolean into a Boolean for proper insertion as a parameter.

Eclipse reports this as an warning.  
![image](https://user-images.githubusercontent.com/55215368/132370205-6021fbf6-2cb6-4f82-90a4-31b1e4c1f989.png)

In past projects, I've had to deal with hard-to-track-down bugs in the code due to misuse of autoboxing.  It's best to avoid it altogether. In fact, I often configure Eclipse to treat these types of warnings as errors to purposely break the compiler if it finds any.

By fixing this autoboxing line, I can have a clean project according to Eclipse.  If I leave it, then it's hard to know which warnings can be ignored and which are real warnings/errors.

Reply:
Interesting Sonar and Intellij are the reverse:
![image](https://user-images.githubusercontent.com/47829178/132372339-04acb9f2-1be3-4020-9939-03e096a0f722.png)

Reply:
LOL, that's funny.  I've always prescribed to https://dzone.com/articles/autoboxing-and-its-pitfalls-1

Reply:
I believe what it is complaining about is the fact the compiler will automatically convert the primitive to the object. So by boxing it you are doing an unnecessary step.

Reply:
That is exactly right, and done very intentionally.  Unfortunately, this is where Eclipse and IntelliJ differs.  Eclipse generates a warning to purposely NOT allow the compiler to automatically convert a primitive to an object.  The Eclipse moto is to never let the compiler do something implicitly when it can be developed explicitly.  Never assume.  However, IntelliJ does the opposite, it has the warning when the it thinks the compiler SHOULD auto convert a primitive to an object.  Both are doing the exact opposite.

I'll disable the Eclipse warning/errors for this project and allow autoboxing everywhere.
Reverted changes to this file.

Reply:
Sonar qube also complains: examineInt(Integer.valueOf(i));  // Noncompliant; boxed int will be auto-unboxed
https://rules.sonarsource.com/java/RSPEC-2153 If it were to be assigning a variable it wouldn't complain but being used inline in a method it does complain. I think the intent is code readability.

Reply:
Right, and that is a great example of unnecessary boxing/unboxing.  That example is taking int i, box to Integer, then unbox to int in the same expression.  Needless.

The logger on this line of code, however, is expecting an Object and being passed a primitive to auto-convert - this is what Eclipse is warning about.

I wish Eclipse could differentiate between unnecessary boxing as in ((int)(Integer)1) and explicit boxing/unboxing.  Unfortunately, there's only the one checkbox - warn for any and all autoboxing, or don't warn.